cdap-common/src/main/java/io/cdap/cdap/common/http/AuthenticationChannelHandler.java

```diff
@@ -86,23 +93,17 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             throw e;
           }
         }
-      } else if (enforceAuthenticatedRequests) {
-        throw new IllegalArgumentException(String.format("Missing Authorization header for request from IP %s",
-                                                         ctx.channel().remoteAddress().toString()));
       }
+      LOG.trace("Got user ID '{}', user IP '{}', and authorization header length '{}'",
+                userID == null ? "NULL" : userID,
+                userIP == null ? "NULL" : userIP,
+                authHeader == null ? "NULL" : String.valueOf(authHeader.length()));
```

Comment:
We should also log the source IP of the channel request using `ctx.channel().remoteAddress().toString()` as was previously logged in line 63 to allow us to better debug where a request came from when it is propagated through multiple services.

Reply:
Done. Added.